### PR TITLE
add general test for hgraph

### PR DESCRIPTION
**Pull request by LHT129**
#40 
- only factory & build/add & general test
- no test for range search

### CODE REVIEW

File: src/index/hgraph_zparameters.cpp [hunk 1]
```diff
@@ -96,7 +96,7 @@ const std::string HGraphParameters::DEFAULT_HGRAPH_PARAMS = format_map(
             "type": "nsw",
             "{GRAPH_PARAMS_KEY}": {
                 "{GRAPH_PARAM_MAX_DEGREE}": 64,
-                "{GRAPH_PARAM_INIT_MAX_CAPACITY}": 1000
+                "{GRAPH_PARAM_INIT_MAX_CAPACITY}": 100
```
Comment (jiaweizone) [inline src/index/hgraph_zparameters.cpp:99]: Why change default value ?

File: tests/fixtures/fixtures.cpp [hunk 2]
```diff
@@ -263,6 +263,24 @@ generate_hnsw_build_parameters_string(const std::string& metric_type, int64_t di
...
+    {{
+        "dtype": "float32",
+        "metric_type": "{}",
+        "dim": {},
+        "index_param": {{
+            "base_quantization_type": "{}"
+        }}
+    }}
+    )";
+    auto build_parameters = fmt::format(parameter_temp, metric_type, dim, base_quantization_type);
```
Comment (jiaweizone) [inline tests/fixtures/fixtures.cpp]: build_parameters -> build_parameters_str ?

File: tests/test_index.cpp [hunk 3]
```diff
@@ -63,7 +63,7 @@ TestIndex::FastGeneralTest(const std::string& name,
     // Test KnnSearch and RangeSearch
     {
         TestKnnSearch(index, dataset, search_parameters, top_k, 0.99);
-        TestRangeSearch(index, dataset, search_parameters, range, 0.99);
+        //        TestRangeSearch(index, dataset, search_parameters, range, 0.99);
```
Comment (jiaweizone) [inline tests/test_index.cpp:66]: Why not run this test ?

File: src/index/hgraph_zparameters.cpp (hunk 1 above)
Comment (LHT129) [reply src/index/hgraph_zparameters.cpp:99]: our test is 1000, to test the resize is right, we should set default count less than 1000

File: tests/test_index.cpp (hunk 3 above)
Comment (LHT129) [reply tests/test_index.cpp:66]: the range implement of the hgraph haven't been approve in PR #163

File: tests/fixtures/fixtures.cpp (hunk 2 above)
Comment (LHT129) [reply tests/fixtures/fixtures.cpp]: done

**Review by jiaweizone — APPROVED** [review]
LGTM

**Review by ShawnShawnYou — APPROVED** [review]

**Pull request merged**